src/auto-evo/simulation/SimulationCache.cs

```diff
@@ -496,4 +508,17 @@ private float CalculateStorageScore(MicrobeSpecies species, BiomeConditions biom
 
         return Math.Clamp(cacheScore, 0, Constants.AUTO_EVO_MAX_BONUS_FROM_ENVIRONMENTAL_STORAGE);
     }
+
+    /// <summary>
+    ///   Calculates cos of angle between the organelle and vertical axis
+    /// </summary>
+    private float CalculateAngleMultiplier(Hex pos)
+    {
+        Vector3 organellePosition = Hex.AxialToCartesian(pos);
+        Vector3 verticalAxis = Hex.AxialToCartesian(new Hex(0, 1));
+        float angleCos = organellePosition.Dot(verticalAxis) / (organellePosition.Length() * verticalAxis.Length());
```

Comment:
What's the math here? I'm having a bit of trouble trying to understand this... Why is 0,1 in polar coordinates converted like this and then multiple lengths are calculated? Why not just do the same as the MovementComponent or `organellePosition.AngleTo(Hex.AxialToCartesian(new Hex(0, 0));`?

Reply:
I changed the maths a little bit so that it is easier to understand. I'm just calculating the cosine of the vector pointing downwards and the vector pointing to the organelle, so that the smaller the angle the bigger the multiplier

Reply:
I guess it is fine now, though there's an unnecessary Normalized call as vector (0, 0, 1) has to be normalized already. Also just to confirm: this will work even if the organelle position is 0,0,0?

Reply:
It will just result as cos(angle) = 0

Reply:
Doesn't that then lead to 0 score? If I remember right the slime jet is biased to go backwards meaning that actual 0,0 position should mean a multiplier of 1...

Reply:
Yes but that is a really rare case and I think adding an if statement would just add some unnecessary calculations which won't result in any noticeable difference in the evolution

Reply:
I would add the if for two reasons:
- Energy calculations impact the player
- And we already have multiple open bug reports about inconsistent stuff when placed at 0,0 (so we'd eventually have someone find out this bug and report it, at which point it will be super hard to diagnose as I won't remember this discussion)

Reply:
You're right, that's an easy fix but I'll do it in the evening cause I don't have access to my laptop rn unfortunately